client/src/pages/Availability/AvailInterface.tsx

```diff
@@ -0,0 +1,154 @@
+import { Typography } from '@mui/material';
+import Grid from '@mui/material/Grid';
+import FormGroup from '@mui/material/FormGroup';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import Checkbox from '@mui/material/Checkbox';
+import Select from '@mui/material/Select';
+import MenuItem from '@mui/material/MenuItem';
+import { Field } from 'formik';
+import { classicNameResolver } from 'typescript';
+
+const Time = () => {
+  const output = [];
+  for (let i = 0; i < 24; i++) {
+    const listItem = <MenuItem value={`${i}:00`}>{`${i}:00`}</MenuItem>;
+    output.push(listItem);
+  }
+  return output;
+};
+
+const createAvailbilityInterface = (day: string, values: any, setFieldValue: any) => (
+  <Grid
+    container
+    columns={3}
+    sx={{
+      height: '4.5rem',
+      display: 'flex',
+      borderBottom: '1px solid #dbdbdb',
+      opacity: values.days[day].active ? 1 : 0.4,
+    }}
+  >
+    <FormGroup sx={{ marginTop: 'auto', marginBottom: 'auto', width: '25%' }}>
+      <FormControlLabel
+        control={
+          <Field
+            id={`days[${day}].active`}
+            name={`days[${day}].active`}
+            checked={values.days[day].active}
+            onChange={(e: any) => {
```

Comment:
Please avoid inline function defs - it impacts code maintainability and even performance.

Ditto where applicable

Reply:
Done the changes

Reply:
Done the changes